RoslynPluginGenerator/NuGet/NuGetMachineWideSettings.cs

```diff
@@ -0,0 +1,54 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="NuGetMachineWideSettings.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+using NuGet;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace SonarQube.Plugins.Roslyn
+{
+    /// <summary>
+    /// Simple implementation of <see cref="IMachineWideSettings"/>.
+    /// Returns any settings in the standard machine-wide NuGet settings folders, together
+    /// with any that exist in the "SonarQube" sub-directory
+    /// </summary>
+    public class NuGetMachineWideSettings : IMachineWideSettings
```

Comment:
NuGet defines an _IMachineWideSettings_ interface that just provides a list of the Settings. We need our own simple  implementation of this to pass to the base _NuGet.Settings.LoadDefaultSettings(...)_ method.